src/ol/geom/circle.js

```diff
@@ -223,7 +227,9 @@ ol.geom.Circle.prototype.setFlatCoordinates =
 
 
 /**
- * Set the radius of the circle. The radius is in the units of the projection.
+ * Set the radius of the circle in projected units.
+ * For EPSG:3857 or EPSG:4326 circles, this method
+ * does not set the radius in meters.
```

Comment:
I think it adds confusion (and mystery) if the bulk of the documentation describes what *does not* happen.

Plus, I don't agree that the units for EPSG:3857 are not meters.

If you feel like documentation needs to be added that explains that circles are only circles in projected coordinates (which holds true to the definition of a circle as points on a plane equidistant from a center point), that could be done as part of a separate pull request.